app/controllers/api_controller/parser.rb

```diff
@@ -41,7 +40,7 @@ def validate_optional_collection_classes
           raise BadRequestError, "Management of #{cname} is unsupported for the Provider class"
         end
       end
-      @collection_klasses[:providers] = "Provider"
+      @collection_klasses[:providers] = Provider
```

Comment:
The same [commit](https://github.com/ManageIQ/manageiq/pull/9718/commits/9f767dac42fb7905c3690a21ba19cf6a813669ba) which changes this, it also amends the `collection_class()` to not call constantize.

Rationale: When I introduced `CollectionConfig.klass` I knew I have two options. Either it returns the class or the caller will always need to remember to run `constantize`. Then I see the collection_klasses being used in same way. Also, I liked this change because it makes the `validate_optional_collection_classes` little more readable for me. It tells me that we store a Provider class for further use, before it was just a string literal to new-comer like me.